paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -194,6 +194,26 @@ public class CoreOptions implements Serializable {
                     .defaultValue(Duration.ofHours(1))
                     .withDescription("The maximum time of completed snapshots to retain.");
 
+    public static final ConfigOption<ExpireExecutionMode> SNAPSHOT_EXPIRE_EXECUTION_MODE =
+            key("snapshot.expire.execution-mode")
```

Comment:
I think about this again. Maybe it is not good to provide async mode now. Maybe there will be some thread conflicts.

The performance maybe good enough now, so we don't need to provide async mode.

Reply:
For a table with a particularly large number of files, simply increasing the number of threads may not be able to delete expired files in time. In our production scenario, a table is divided into partitions by day, and 10,000+ files are generated each commit, and each partition has about 5 million files. When the partition expires, it is difficult to delete all within the checkpoint interval.